Framework/Algorithms/src/GetAllEi.cpp

```diff
@@ -383,8 +384,7 @@ void GetAllEi::exec() {
   std::sort(peaks.begin(), peaks.end());
 
   // finalize output
-  auto result_ws = API::WorkspaceFactory::Instance().create("Workspace2D", 1,
-                                                            nPeaks, nPeaks);
+  API::MatrixWorkspace_sptr result_ws = create<Workspace2D>(1, Points(nPeaks));
```

Comment:
Just FYI: You can also use `auto` instead of givin the type youself (`API::MatrixWorkspace_sptr`). The only consequence is that you need to use `std::move(result_ws)` when calling `setProperty("OutputWorkspace", ...)` below.